File: modules/credential-manager/android/src/main/java/expo/modules/credentialmanager/CredentialManagerModule.kt
package expo.modules.settings

import android.content.Context
import android.content.SharedPreferences
import androidx.core.os.bundleOf
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import expo.modules.kotlin.types.Enumerable

class CredentialManagerModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("CredentialManager")

    Events("onChange")

    Function("setTheme") { theme: Theme ->
      getPreferences().edit().putString("theme", theme.value).commit()
      this@CredentialManagerModule.sendEvent("onChange", bundleOf("theme" to theme.value))
    } 

    Function("getTheme") {
      return@Function getPreferences().getString("theme", Theme.SYSTEM.value)
    }
  }

  private val context
  get() = requireNotNull(appContext.reactContext)

  private fun getPreferences(): SharedPreferences {
    return context.getSharedPreferences(context.packageName + ".credentialmanager", Context.MODE_PRIVATE)
  }
}

enum class Theme(val value: String) : Enumerable {
  LIGHT("light"),
  DARK("dark"),
  SYSTEM("system")
}

File: modules/credential-manager/ios/CredentialManagerModule.swift
import ExpoModulesCore

public class CredentialManagerModule: Module {
  public func definition() -> ModuleDefinition {
    Name("CredentialManager")

    Events("onChange")

    Function("setTheme") { (theme: Theme) -> Void in
      UserDefaults.standard.set(theme.rawValue, forKey:"theme")
      sendEvent("onChange", [
        "theme": theme.rawValue
      ])
    }

    Function("getTheme") { () -> String in
      UserDefaults.standard.string(forKey: "theme") ?? Theme.system.rawValue
    }
  }

  enum Theme: String, Enumerable {
    case light
    case dark
    case system
  }
}
